feat: refactor and add accessor protocol support to `stats/base/nanvariancetk`

prajjwalbajpai opened the pull request:
Resolves #5676.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds support for accessor arrays and refactor `stats/base/nanvariancetk`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5676

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

prajjwalbajpai committed:
feat: refactor and add protocol support to stats/base/nanvariancetk

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

prajjwalbajpai committed:
fix: added NaN to examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

prajjwalbajpai commented:
@aayush0325 please review

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancetk/docs/repl.txt:
```diff
@@ -1,10 +1,10 @@
 
-    The `N` and `stride` parameters determine which elements in `x` are accessed
-    at runtime.
+    The `N` and `stride` parameters determine which elements
```
update line wrapping

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancetk/lib/main.js:
```diff
@@ -20,14 +20,31 @@
+*
+* @example
+* var x = [ 1.0, -2.0, NaN, 2.0 ];
+* var N = x.length;
```
```suggestion
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancetk/lib/main.js:
```diff
@@ -20,14 +20,31 @@
+* var x = [ 1.0, -2.0, NaN, 2.0 ];
+* var N = x.length;
+*
+* var v = nanvariancetk( N, 1, x, 1 );
```
```suggestion
* var v = nanvariancetk( 4, 1, x, 1 );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancetk/test/test.main.js:
```diff
@@ -148,6 +202,60 @@ tape( 'the function calculates the sample variance of a strided array (ignoring
 	t.end();
 });
 
+tape( 'the function calculates the sample variance of a strided array (ignoring `NaN` values) (accessors)', function test( t ) {
```
```suggestion
tape( 'the function calculates the sample variance of a strided array (accessors)', function test( t ) {
```
kindly update these test messages everywhere else in this PR

aayush0325 reviewed (changes requested):
minor comments, once addressed this PR will be in really good shape

prajjwalbajpai committed:
fix: changed wrapping and more

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

prajjwalbajpai commented:
@aayush0325 done

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancetk/README.md:
```diff
@@ -98,7 +98,7 @@ The use of the term `n-1` is commonly referred to as Bessel's correction. Note,
-#### nanvariancetk( N, correction, x, stride )
+#### nanvariancetk( N, correction, x, strideX )
 
 Computes the [variance][variance] of a strided array `x` ignoring `NaN` values and using a one-pass textbook algorithm.
```
```suggestion
Computes the [variance][variance] of a strided array ignoring `NaN` values and using a one-pass textbook algorithm.
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM. Thanks, @aayush0325, for reviewing.
